Makefile

```diff
@@ -102,8 +102,9 @@ lint: lint-docs ## Check the source files for syntax and format issues
 test: ## run the test-harness
 	test/test-harness.sh
 
-# include only variables which have a defined value
-PACKER_VARIABLES := $(foreach packerVar,$(AVAILABLE_PACKER_VARIABLES),$(if $($(packerVar)),$(packerVar)))
+# include only variables which have a defined value or the variable is imds_support
+PACKER_IMDS_SUPPORT_VAR := imds_support
```

Comment:
What's the intention of this?

Reply:
> What's the intention of this?

The valid values for `imds_supports` are "v2" and "".

Currently, the behavior is, when a packer value is assigned with "" (empty value), it is a noop.

However, we want to change this behavior for `imds_support`, in case of `imds_support=""`, it should be treated as a valid assignment to disable IMDSv2 on the AMI.

Reply:
Did you try it without this extra logic in the Makefile? I think it would work the way you intend. If the template variable has the value of `""`, then `{{user `imds_support`}}` is going to resolve to the empty string as expected. We set the default value of `""` in our default variable file. A user doesn't need to additionally define `imds_support` in their `make` command or in their own variable file, because our default would apply.

Reply:
@cartermckinnon 
> If a user wants to leave IMDSv1 enabled, they don't need to define imds_support in their make command or in their own variable file, because our default would apply.
This is correct

However, imagine that we will make IMDSv2 enabled by default in the future, and the customer wants to use IMDSv1 for whatever reason, it won't work without this change.
When it sees `make k8s imds_support=''`, it won't have the effect to set to IMDSv1, it will just ignore this, it is a noop (`''` and `'v2.0'` are the only 2 valid values for `imds_support`).

Reply:
I think we’re unlikely to need that flexibility. We don’t ever have to change the default, we could just always pass v2.0 for official builds. After a while, we can remove the variable altogether and hardcode v2.0 in the template.

Reply:
@cartermckinnon Makefile change reverted